### ISSUE TITLE
Simple join of `baseUrl` and `path` may result in invalid URLs

### ISSUE THREAD

**Issue by rdmurphy** [drwpow/openapi-typescript#1150]
**Description**

Hello! I noticed that when you have a `baseUrl` with a trailing slash (e.g. `https://example.com/api/`) and `paths` with a leading slash (e.g. `/create-post`) it results in an invalid URL/request. If I am reading the code correctly I think the problem surfaces in this line:

https://github.com/drwpow/openapi-typescript/blob/65ff4b1826f1a70bce2acf5cd0ba77ca18b08502/packages/openapi-fetch/src/index.ts#L69

I think it assumes those two strings can always just be joined directly. If this is the case then I think [the README is also suggesting this would work when it may not](https://github.com/drwpow/openapi-typescript/blob/65ff4b1826f1a70bce2acf5cd0ba77ca18b08502/packages/openapi-fetch/README.md?plain=1#L17-L25).

**Reproduction**

Here is a tiny types file for making a request to `https://jsonplaceholder.typicode.com`:

```ts
/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */


export interface paths {
  "/posts": {
    get: {
      responses: {
        /** @description OK */
        200: {
          content: {
            "application/json": ({
                userId?: number;
                id?: number;
                title?: string;
                body?: string;
              })[];
          };
        };
      };
    };
  };
}
```

And the script I ran:

```ts
// I'm using Deno, but it's the same otherwise
import createClient from "https://esm.sh/openapi-fetch@0.2.1";

import type { paths } from "./api.d.ts";

const client = createClient<paths>({
  baseUrl: "https://jsonplaceholder.typicode.com/", // note the trailing slash
});

const posts = await client.get("/posts", {});
```

**Expected result**

I think it'd be helpful if `openapi-fetch` smoothed this over! Maybe the leading slash of a `path` should be removed pre-joining if the `baseUrl` has a trailing slash. I could also see this using the [`URL()` constructor instead](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) to get a proper join for free.

**Checklist**

- [x] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-fetch/blob/main/CONTRIBUTING.md))

**Comment by drwpow** [comment]
Ah you know what? Since OpenAPI requires that all your URLs start with a `/`, then we could reasonably assume that all requests will, too, and that does mean that `baseUrl` should have the trailing slash stripped.

If this were any fetch library I’d be hesitant of making assumptions of how URLs are formed (even the `URL()` constructor allows you to have `//` in the middle of a pathname), but you’re right we can lean on the OpenAPI spec this enforces.

As a sidenote, we can’t use the `URL()` constructor method because `baseUrl` doesn’t have to be a full domain—some people are fetching locally (e.g. `{ baseUrl: '/api/v1' }`).

**Comment by drwpow** [comment]
This is now fixed in `0.3.0`. No stripping of leading slashes in the URL is done, because that should be enforced already by the OpenAPI typing
